system/Model.php

```diff
@@ -498,6 +498,47 @@ public function first()
 		return $eventData['data'];
 	}
 
+    	//--------------------------------------------------------------------
+
+    	/**
+     	* Returns the last row of the result set. Will take any previous
+     	* Query Builder calls into account when determining the result set.
+     	*
+     	* @return array|object|null
+     	*/
+    	public function last()
+    	{
+		$builder = $this->builder();
+
+		if ($this->tempUseSoftDeletes === true)
+		{
+	    		$builder->where($this->table . '.' . $this->deletedField, null);
+		}
+		else
+		{
+			if ($this->useSoftDeletes === true && empty($builder->QBGroupBy) && ! empty($this->primaryKey))
+			{
+				$builder->groupBy($this->table . '.' . $this->primaryKey);
+			}
+		}
+
+		// Some databases, like PostgreSQL, need order
+		// information to consistently return correct results.
+		if (! empty($builder->QBGroupBy) && empty($builder->QBOrderBy) && ! empty($this->primaryKey))
+		{
+	    		$builder->orderBy($this->table . '.' . $this->primaryKey, 'asc');
+		}
+
+		$eventData = $this->trigger('afterFind', [
+	    		'data' => $builder->get()->getLastRow($this->tempReturnType)
```

Comment:
Actually, I just have a thought that we can make it better this way:

```suggestion
		$subQuery   = $builder->getCompiledSelect();
		$subBuilder = $this->db->table('(' . $subQuery . ') subQuery');
		$row = $subBuilder->orderBy('subQuery.' . $this->primaryKey, 'desc')->limit(1, 0)->get()->getFirstRow($this->tempReturnType);
                
		$eventData = $this->trigger('afterFind', [
	    		'data' => $row
```
Any other ideas?